text/0039-reduce-module-size.md

```diff
@@ -0,0 +1,518 @@
+# Reduce aws-cdk-lib package size
+
+* **Original Author(s):**: @madeline-k
+* **Tracking Issue**: #39
+* **API Bar Raiser**: tbd
+
+The AWS CDK v2 framework library, `aws-cdk-lib` contains all packages from AWS
+CDK v1 in one monolithic package. While this is a great value for customers in
+making dependency management much easier, it has also resulted in increasing the
+unpacked npm package size to 238 MB, at the time of writing. This feature
+reduces the size of the package to 100 MB.
```

Comment:
Do we have a section that breaks down, with all the proposed changes, what we will get the size down to?

Reply:
Not precisely. It is mentioned in the execution plan section at the end. I don't want to make this super concrete, because we haven't done enough prototyping to be super confident in these estimates. So far, we can be reasonably confident that the proposed solutions can reduce the size by more than 100 MB. But it's hard to make a reliable estimate of future size, especially since we will keep adding more code and features to the library before the solutions proposed here are implemented.